services/ymax-planner/src/support.ts

```diff
@@ -1,22 +1,107 @@
-import { WebSocketProvider, Log, type Filter } from 'ethers';
+import { WebSocketProvider, Log } from 'ethers';
+import type { Filter } from 'ethers';
 import type { CaipChainId } from '@agoric/orchestration';
-import { objectMap } from '@agoric/internal';
+import type { ClusterName } from '@agoric/client-utils';
+import { fromTypedEntries, objectMap, typedEntries } from '@agoric/internal';
 import {
+  CaipChainIds,
   EvmWalletOperationType,
   YieldProtocol,
 } from '@agoric/portfolio-api/src/constants.js';
-import type { ClusterName } from './config.ts';
+import type { SupportedChain } from '@agoric/portfolio-api/src/constants.js';
+import type {
+  PoolKey as InstrumentId,
+  PoolPlaceInfo,
+} from '@aglocal/portfolio-contract/src/type-guards.js';
+import {
+  aaveRewardsControllerAddresses,
+  compoundAddresses,
+} from '@aglocal/portfolio-deploy/src/axelar-configs.js';
 import type { EvmContext } from './pending-tx-manager.ts';
 
+type ROPartial<K extends string, V> = Readonly<Partial<Record<K, V>>>;
+
 const { entries } = Object;
 
 type HexAddress = `0x${string}`;
 
+/**
+ * @deprecated should come from e.g. @agoric/portfolio-api/src/constants.js
+ *   or @agoric/orchestration
+ */
 export type UsdcAddresses = {
   mainnet: Record<CaipChainId, HexAddress>;
   testnet: Record<CaipChainId, HexAddress>;
 };
 
+const spectrumChainIdFromCaipChainId = (caipChainId: CaipChainId) => {
```

Comment:
it's a small fixed mapping. please use an explicit lookup table. that makes string search possible and we'll have to do that anyway for their ENUM change.

Reply:
By "explicit" I meant with string literals (for the reason to make "string search possible" and that "we'll have to do [the literal-based mapping] anyway for their ENUM change)

Reply:
Done.